pandas/core/arrays/interval.py

```diff
@@ -547,6 +550,58 @@ def __setitem__(self, key, value):
         right.values[key] = value_right
         self._right = right
 
+    def __eq__(self, other):
+        # ensure pandas array for list-like and eliminate non-interval scalars
+        if is_list_like(other):
```

Comment:
this particular check can be pushed to the base class

Reply:
I don't quite follow. Whose the base class here? ExtensionArray?

Reply:
yes